src/util/transition/ngbFadingTransition.ts

```diff
@@ -3,3 +3,21 @@ import {NgbTransitionStartFn} from './ngbTransition';
 export const ngbAlertFadingTransition: NgbTransitionStartFn = ({classList}: HTMLElement) => {
   classList.remove('show');
 };
+
+export interface NgbToastCtx { direction: 'show' | 'hide'; }
+
+export const ngbToastFadeInTransition: NgbTransitionStartFn<NgbToastCtx> = ({classList}: HTMLElement) => {
+  classList.remove('hide');
+  classList.add('showing');
+
+  return () => {
+    classList.remove('showing');
+    classList.add('show');
+  };
+};
+
+export const ngbToastFadeOutTransition: NgbTransitionStartFn<NgbToastCtx> = ({classList}: HTMLElement) => {
+  classList.remove('show');
+
+  return () => { classList.add('hide'); };
+};
```

Comment:
After 3 widgets conversion, I am still not convinced about the need for this abstraction.
These functions should be inlined in each component, even if used several times.

Reply:
It's a matter of coding preferences. Even if simple, I prefer to abstract this specific action done in the widget. 

`ngbToastFadeOutTransition` is a kind of documentation. When I read `
this._element.nativeElement, ngbToastFadeOutTransition, {animation: this.animation, runningTransition: 'stop'});` in the toast, I can easily understand what the code intend to do (a fade out), and separate the fade out code form the rest of the method code. In the fade out code, I can also easily know what are the Bootstrap specs on this transition.

But it's up to the main project decision and guidelines. No worries, I can change it if needed ;)

Reply:
Added this to the list → https://github.com/ng-bootstrap/ng-bootstrap/issues/3747#issue-626345524